src/features/inlay_hints.zig

```diff
@@ -188,29 +189,82 @@ fn writeBuiltinHint(builder: *Builder, parameters: []const Ast.Node.Index, argum
         const colonIndex = std.mem.indexOfScalar(u8, arg, ':');
         const type_expr: []const u8 = if (colonIndex) |index| arg[index + 1 ..] else &.{};
 
-        var label: ?[]const u8 = null;
+        var maybe_label: ?[]const u8 = null;
         var no_alias = false;
         var comp_time = false;
 
         var it = std.mem.splitScalar(u8, arg[0 .. colonIndex orelse arg.len], ' ');
         while (it.next()) |item| {
             if (item.len == 0) continue;
-            label = item;
+            maybe_label = item;
 
             no_alias = no_alias or std.mem.eql(u8, item, "noalias");
             comp_time = comp_time or std.mem.eql(u8, item, "comptime");
         }
 
+        const label = maybe_label orelse return;
+        if (label.len == 0 or std.mem.eql(u8, label, "...")) return;
+
         try builder.appendParameterHint(
             tree.firstToken(parameter),
-            label orelse "",
+            label,
             std.mem.trim(u8, type_expr, " \t\n"),
             no_alias,
             comp_time,
         );
     }
 }
 
+/// Takes a variable declaration AST node. If the type is inferred, attempt to infer it and display it as a hint.
+fn writeVariableDeclHint(builder: *Builder, decl_node: Ast.Node.Index) !void {
+    const tracy_zone = tracy.trace(@src());
+    defer tracy_zone.end();
+
+    const handle = builder.handle;
+    const tree = handle.tree;
+
+    var type_references = Analyser.ReferencedType.Set.init(builder.arena);
+    var reference_collector = Analyser.ReferencedType.Collector.init(&type_references);
+
+    const decl_name = Analyser.getContainerDeclName(
+        tree,
+        null,
+        decl_node,
+    ) orelse return;
+    const source_index = offsets.tokenToIndex(tree, tree.firstToken(decl_node));
+    const decl_handle = try builder.analyser.lookupSymbolGlobal(
+        handle,
+        decl_name,
+        source_index,
+    ) orelse return;
+
+    var type_str: []const u8 = "unknown";
+    if (try decl_handle.resolveType(builder.analyser)) |resolved_type| {
+        try builder.analyser.referencedTypes(
+            resolved_type,
+            &type_str,
+            &reference_collector,
+        );
+    }
```

Comment:
You can probably achieve the same with `resolveTypeOfNode`.

```suggestion
    const resolved_type = (try builder.analyser.resolveTypeOfNode(.{ .node = decl_node, .handle = handle })) orelse return;

    var type_references = Analyser.ReferencedType.Set.init(builder.arena);
    var reference_collector = Analyser.ReferencedType.Collector.init(&type_references);

    var type_str: []const u8 = "unknown";
    try builder.analyser.referencedTypes(
        resolved_type,
        &type_str,
        &reference_collector,
    );
```